File: LeetCode-Programs/Partition Array Into Three Parts With Equal Sum.py
class Solution:
    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
        total_sum=sum(arr)
        temp= total_sum//3
        summ, count=0, 0
        for i in range(len(arr)):
            summ+=arr[i]
            if summ==temp:
                count+=1
                summ=0
        return count>=3 and total_sum%3==0

File: LeetCode-Programs/Partition Array Into Three Parts With Equal Sum.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& arr) {        
        int total_sum= accumulate(arr.begin(), arr.end(), 0);
        int temp = total_sum/3;
        int sum=0, count=0;
        for(int i=0; i<arr.size(); i++){
            sum+=arr[i];
            if(sum==temp){
                count++;
                sum=0;
            }
        }
        return count>=3 && total_sum %3==0;
    }
};
